src/prefect/agent/kubernetes/agent.py

```diff
@@ -366,9 +369,19 @@ def generate_job_spec_from_environment(
         # Use image pull secrets if provided
         image_pull_secrets = os.getenv("IMAGE_PULL_SECRETS")
         if image_pull_secrets:
-            job["spec"]["template"]["spec"]["imagePullSecrets"][0][
-                "name"
-            ] = image_pull_secrets
+            secrets = image_pull_secrets.split(",")
+            for idx, secret_name in enumerate(secrets):
+                # this check preserves behavior from previous releases,
+                # where prefect would only overwrite the first entry in
+                # imagePullSecrets
+                if idx == 0:
+                    job["spec"]["template"]["spec"]["imagePullSecrets"][0] = {
+                        "name": secret_name
+                    }
```

Comment:
@jcrist just curious why we do this

Reply:
The old way of managing the agent made a lot of assumptions on the underlying template (for better or worse). This will all go away with the new `KubernetesRun` pattern (once everything else is deprecated), but we can't break compatibility for now.